### PR TITLE
fix: Show REST-specific examples in markdown docs for REST-only clients

### LINKED ISSUE #622: REST libraries code examples in `Readme.md` and `Authentication.md` are incorrect
REST generation generates non-rest code samples for in `{Readme, Authentication}.md` files for the libraries.
Can be seen here (README.md, initializing the client and making a request) https://github.com/googleapis/gapic-generator-ruby/blob/master/shared/output/cloud/compute_small/README.md#quick-start
and here (AUTHENTICATION.md, initializing the client) https://github.com/googleapis/gapic-generator-ruby/blob/master/shared/output/cloud/compute_small/AUTHENTICATION.md#quickstart

Also the logging examples should be turned off if the library is rest-only since they are talking about GRPC

### PR DESCRIPTION
**Pull request by dazuma**
Closes #622

### CODE REVIEW

File: gapic-generator/lib/gapic/presenters/service_rest_presenter.rb [hunk 1]
```diff
@@ -127,6 +127,30 @@ def test_client_file_path
...
+      def configure_client_call
+        "#{client_name_full}.configure"
+      end
+
+      ##
+      # The method to use for quick start samples. Normally this is simply the
+      # first non-client-streaming method defined, but it can be overridden via
+      # a gem config.
+      #
+      # @return [Gapic::Presenters::MethodRestPresenter]
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/presenters/service_rest_presenter.rb:149]: , nil]

File: gapic-generator/lib/gapic/presenters/service_rest_presenter.rb [hunk 2]
```diff
@@ -127,6 +127,30 @@ def test_client_file_path
         main_service.service_file_path.sub ".rb", "_test.rb"
       end
 
+      ##
+      # @return [String]
+      #
+      def credentials_class_xref
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/presenters/service_rest_presenter.rb:133]: Is this used anywhere?

File: gapic-generator/lib/gapic/presenters/gem_presenter.rb [hunk 3]
```diff
@@ -239,6 +239,10 @@ def quick_start_service
         result || first_non_common_service
       end
 
+      def show_grpc_logging_docs?
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/presenters/gem_presenter.rb:242]: wtb yardoc

File: gapic-generator/lib/gapic/presenters/method_rest_presenter.rb [hunk 4]
```diff
@@ -189,6 +189,24 @@ def query_string_params_var_name
...
       end
+
+      ##
+      # Method name
+      #
+      # @return [String]
+      #
+      def name
+        @main_method.name
+      end
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/presenters/method_rest_presenter.rb:200]: Nit: replace `@main_method.name`, e.g. in `transcoding_helper_name` with this

**Review by viacheslav-rostovtsev — APPROVED** [review]
Looking good with some minor notes

File: gapic-generator/lib/gapic/presenters/service_rest_presenter.rb (hunk 2 above)
Comment (dazuma) [reply gapic-generator/lib/gapic/presenters/service_rest_presenter.rb:133]: Yes, it's used in the cloud's authentication.erb template.

File: gapic-generator/lib/gapic/presenters/service_rest_presenter.rb (hunk 1 above)
Comment (dazuma) [reply gapic-generator/lib/gapic/presenters/service_rest_presenter.rb]: done

File: gapic-generator/lib/gapic/presenters/gem_presenter.rb (hunk 3 above)
Comment (dazuma) [reply gapic-generator/lib/gapic/presenters/gem_presenter.rb:249]: done

File: gapic-generator/lib/gapic/presenters/method_rest_presenter.rb (hunk 4 above)
Comment (dazuma) [reply gapic-generator/lib/gapic/presenters/method_rest_presenter.rb:200]: done

**Pull request merged**